test/mbed_gt_cli.py

```diff
@@ -38,14 +45,50 @@ def test_get_greentea_version(self):
         self.assertEqual(b.isdigit(), True)
         self.assertEqual(c.isdigit(), True)
 
-    def get_hello_string(self):
+    def test_print_version(self):
+        version = mbed_greentea_cli.get_greentea_version()
+
+        sys.stdout = stdout_capture = StringIO()
+        mbed_greentea_cli.print_version()
+        sys.stdout = sys.__stdout__
+
+        printed_version = stdout_capture.getvalue().splitlines()[0]
+        self.assertEqual(printed_version, version)
+
+    def test_get_hello_string(self):
         version = mbed_greentea_cli.get_greentea_version()
         hello_string = mbed_greentea_cli.get_hello_string()
 
         self.assertIs(type(version), str)
         self.assertIs(type(hello_string), str)
         self.assertIn(version, hello_string)
 
+    def test_get_local_host_tests_dir_invalid_path(self):
+        test_path = mbed_greentea_cli.get_local_host_tests_dir("invalid-path")
+        self.assertEqual(test_path, None)
+
+    def test_get_local_host_tests_dir_valid_path(self):
+        path = "."
+        test_path = mbed_greentea_cli.get_local_host_tests_dir(path)
+        self.assertEqual(test_path, path)
+
+    def test_get_local_host_tests_dir_default_path(self):
+        import os
+        import shutil
+        import tempfile
+
+        curr_dir = os.getcwd()
+        test1_dir = tempfile.mkdtemp()
+        test2_dir = os.mkdir(os.path.join(test1_dir, "test"))
+        test3_dir = os.mkdir(os.path.join(test1_dir, "test", "host_tests"))
+
+        os.chdir(test1_dir)
+
+        test_path = mbed_greentea_cli.get_local_host_tests_dir("")
+        self.assertEqual(test_path, "./test/host_tests")
+
+        os.chdir(curr_dir)
```

Comment:
Why `os.chdir` here? Why not pass the `test1_dir` to `get_local_host_tests_dir`?

Reply:
That tests a different branch which is covered by the previous test. Passing an empty string asserts that the function checks that `./test/host_tests` exists.

Reply:
Thanks for the explanation. 👍